scripts/inference/ft/mpt_ckpt_to_ft.py

```diff
@@ -0,0 +1,294 @@
+            save_path = save_dir + '/model.' + key + '.%d.bin' % (i * factor +
+                                                                  j)
+            split_vals[j].tofile(save_path)
+            if config['no_bias']:
+                write_empty_bias(key, save_path, (3, split_vals[j].shape[-1]))
+
+    else:
+        print("[ERROR] cannot find key '{}'".format(key))
+
+
+def split_and_convert(args):
+    save_dir = args.save_dir + '/%d-gpu/' % args.infer_gpu_num
+
+    if (os.path.exists(save_dir) == False):
+        os.makedirs(save_dir)
+
+    factor = args.infer_gpu_num
+
+    # do conversion on cpu
+    torch_device = 'cpu'
+
+    model = transformers.AutoModelForCausalLM.from_pretrained(
+        args.name_or_dir, trust_remote_code=True).to(torch_device)
+
+    hf_config = vars(model.config)
+
+    config = configparser.ConfigParser()
+    config['gpt'] = {}
+    try:
+        config['gpt']['model_name'] = 'mpt' if hf_config[
+            '_name_or_path'] == '' else hf_config['_name_or_path']
+        config['gpt']['head_num'] = str(hf_config['n_heads'])
+        n_embd = hf_config['d_model']
+        config['gpt']['size_per_head'] = str(n_embd // hf_config['n_heads'])
+        config['gpt']['inter_size'] = str(n_embd * hf_config['expansion_ratio'])
+        config['gpt']['max_pos_seq_len'] = str(hf_config['max_seq_len'])
+        config['gpt']['num_layer'] = str(hf_config['n_layers'])
+        config['gpt']['vocab_size'] = str(hf_config['vocab_size'])
+        config['gpt']['start_id'] = str(hf_config['bos_token_id'])
+        config['gpt']['end_id'] = str(hf_config['eos_token_id'])
+        config['gpt']['weight_data_type'] = args.weight_data_type
+        config['gpt']['tensor_para_size'] = str(args.infer_gpu_num)
+        if hf_config['attn_config']['alibi']:
+            config['gpt']['has_positional_encoding'] = str(False)
+            config['gpt']['use_attention_linear_bias'] = str(True)
+
+        with open(save_dir + '/config.ini', 'w') as configfile:
+            config.write(configfile)
+    except:
+        print(f'Fail to save the config in config.ini.')
+
+    np_weight_data_type = get_weight_data_type(args.weight_data_type)
+
+    param_remapping = {
+        'norm_1.bias': 'input_layernorm.bias',
+        'norm_1.weight': 'input_layernorm.weight',
+        'attn.Wqkv.bias': 'attention.query_key_value.bias',
+        'attn.Wqkv.weight': 'attention.query_key_value.weight',
+        'attn.out_proj.bias': 'attention.dense.bias',
+        'attn.out_proj.weight': 'attention.dense.weight',
+        'norm_2.bias': 'post_attention_layernorm.bias',
+        'norm_2.weight': 'post_attention_layernorm.weight',
+        'ffn.up_proj.bias': 'mlp.dense_h_to_4h.bias',
+        'ffn.up_proj.weight': 'mlp.dense_h_to_4h.weight',
+        'ffn.down_proj.bias': 'mlp.dense_4h_to_h.bias',
+        'ffn.down_proj.weight': 'mlp.dense_4h_to_h.weight',
+    }
+
+    pool = multiprocessing.Pool(args.processes)
```

Comment:
Is the complexity of multiprocessing needed? How much does this speed up the conversion?

Reply:
@hanlint : Removed multiprocessing. As implemented it wasn't doing anything. Currently it takes ~5 mins to convert the 7B checkpoint.